payments/models.py

```diff
@@ -0,0 +1,369 @@
+from datetime import datetime, timedelta
+from decimal import Decimal
+
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from django.core.validators import MinValueValidator
+from django.db import models
+from django.db.models import OuterRef, Q, Subquery
+from django.utils import translation
+from django.utils.formats import localize
+from django.utils.functional import cached_property
+from django.utils.timezone import now, utc
+from django.utils.translation import ugettext_lazy as _
+from rest_framework import serializers
+
+from resources.models import Reservation, Resource
+from resources.models.utils import generate_id
+
+from .exceptions import OrderStateTransitionError
+from .utils import convert_aftertax_to_pretax, get_price_period_display, rounded
+
+# The best way for representing non existing archived_at would be using None for it,
+# but that would not work with the unique_together constraint, which brings many
+# benefits, so we use this sentinel value instead of None.
+ARCHIVED_AT_NONE = datetime(9999, 12, 31, tzinfo=utc)
```

Comment:
Hmm, how does that prevent multiple products with `archived_at` `null` from existing?

But yeah, the current solution isn't *that* ugly, and can be implemented with Django code only, so I guess it ok :)

Reply:
Oops, it should read `IS NULL` instead of `IS NOT NULL` :D

Reply:
Ah ok, makes much more sense now :D Yup, that seems like a nice solution indeed. Actually, it might be even possible to pull that off in Django 2.2 using https://docs.djangoproject.com/en/2.2/ref/models/indexes/#condition